audb/core/load.py

```diff
@@ -746,6 +656,58 @@ def job(table):
     )
 
 
+def filtered_dependencies(
+        name: str,
+        version: str,
+        media: typing.Union[str, typing.Sequence[str]],
+        tables: typing.Union[str, typing.Sequence[str]],
+        cache_root: str = None,
+) -> pd.DataFrame:
+    r"""Filter media by tables.
+
+    Return all media files from ``media``
+    that are mentioned in at least one table
+    from ``tables``.
+    This has to download every table
+    for doing the check.
+
+    Args:
+        name: name of database
+        version: version of database
+        media: media files
+        tables: table IDs
+        cache_root: cache folder where databases are stored.
+            If not set :meth:`audb.default_cache_root` is used
+
+    Returns:
+        filtered dependencies
+
+    """
+    deps = dependencies(name, version=version, cache_root=cache_root)
+    if tables is None and media is None:
+        df = deps._df
```

Comment:
```suggestion
        df = deps()
```

In case we someday decide to switch back to a dictionary inside `Dependencies` :)